IPython/html/static/widgets/js/widget_bool.js

```diff
@@ -129,10 +129,11 @@ define([
 
                 var description = this.model.get('description');
                 this.$el.attr("title", this.model.get("tooltip"));
-                if (description.trim().length === 0) {
+                this.$el.text(description);
+                var icon = this.model.get("icon");
+                $('<i class="fa"></i>').prependTo(this.$el).addClass(icon);
+                if (description.trim().length === 0 && icon.trim().length ===0) {
                     this.$el.html("&nbsp;"); // Preserve button height
-                } else {
-                    this.$el.text(description);
```

Comment:
It seems like this else block still makes sense. This change currently does $el.text unconditionally, even though the if statement here will override the result.